### PR TITLE
fix(meta): set max encode/decode size to 16MB for meta-grpc and raft client/server

### PR DESCRIPTION
**Pull request by drmingdrmer**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

##### fix(meta): set max encode/decode size to 16MB for meta-grpc and raft client/server

- Part of: #11338

## Related Issues

### CODE REVIEW

File: src/meta/client/src/grpc_client.rs [hunk 1]
```diff
@@ -542,9 +544,12 @@ impl MetaGrpcClient {
                     .await;
                     match new_token {
                         Ok(token) => {
-                            return Ok(MetaServiceClient::with_interceptor(c, AuthInterceptor {
-                                token,
-                            }));
+                            let client =
+                                MetaServiceClient::with_interceptor(c, AuthInterceptor { token })
+                                    .max_decoding_message_size(16 * 1024 * 1024)
```
Comment (sundy-li) [inline src/meta/client/src/grpc_client.rs:549]: The default limit is `4G`, do you want to make it smaller?

Comment (drmingdrmer) [reply src/meta/client/src/grpc_client.rs:549]: The encoding limit is the max size a client can send and is 4GB, the decoding limit is the max size a server can receive and is 4MB.

File: src/meta/client/src/grpc_client.rs [hunk 2]
```diff
@@ -530,7 +530,9 @@ impl MetaGrpcClient {
             let channel = self.make_channel(Some(addr)).await;
             match channel {
                 Ok(c) => {
-                    let mut client = MetaServiceClient::new(c.clone());
+                    let mut client = MetaServiceClient::new(c.clone())
+                        .max_decoding_message_size(16 * 1024 * 1024)
```
Comment (ariesdevil) [inline src/meta/client/src/grpc_client.rs:534]: This arithmetic expression appears many times and is recommended as a constant.

Comment (drmingdrmer) [reply src/meta/client/src/grpc_client.rs]: Done. my man

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**